### ISSUE TITLE
path in log messages not always shown and not shown at all if in indirection

### ISSUE THREAD

**Issue by jipclaassens** [ObjectVision/GeoDMS#795]
```
attribute<rdc_meter> geometry_buffer (poly) := bg_buffer_multi_polygon(geometry,Buffer_size[float64], 8b);
```

![image](https://github.com/user-attachments/assets/19bbba92-48e0-4a60-9bbf-41ed767c2b6a)

```
attribute<rdc_meter> geometry_buffer_ind (poly) :=
	= 'bg_buffer_multi_polygon(
		geometry,
		max_elem(
			VariantParameters/VariantK/AfstandGeluidsnorm[VariantParameters/VariantK/V/'+Variant_name+'],
			CaseClassifications/WindTurbineType/Ashoogte + 0.5f * CaseClassifications/WindTurbineType/RotorDiameter,
			CaseClassifications/WindTurbineType/MaxWerpAfstandNom)[float64],
	8b)';
```

![image](https://github.com/user-attachments/assets/26ffc81f-990e-4218-9b14-1e48fa519a8d)

-----

OVSRV08
`C:\ProjDir\_issues\RSopen_issue795\cfg\main.dms`

Activate one or the other. Alt+r between attributes, otherwise it will be instantly done and no messages shown at all.
```
/VariantData/Wind_restricties/Dynamisch/kwestbare_gebouwen/Woningen/geometry_buffer
/VariantData/Wind_restricties/Dynamisch/kwestbare_gebouwen/Woningen/geometry_buffer_ind
```

**Comment by jipclaassens** [comment]
![image](https://github.com/user-attachments/assets/1c1d4406-470e-4dc6-adf8-cf9d5a2db637)

**Comment by eoudejans** [comment]
Related to #763

**Comment by jipclaassens** [comment]
Cannot reproduce:

![Image](https://github.com/user-attachments/assets/02609fca-5882-44d9-b79f-b633e6f45b89)

**Comment by jipclaassens** [comment]
Both case, no path:

![Image](https://github.com/user-attachments/assets/91525677-03a1-433b-a53b-0a7e67fc8cdc)

![Image](https://github.com/user-attachments/assets/d89dca11-ccd8-43bb-9b24-9d206b927e50)

Reproducable on OVSRV06

**Comment by MaartenHilferink** [comment]
@jipclaassens  Is dit met #1188 nu ook opgelost?

**Comment by jipclaassens** [comment]
No, #1188 was a different mechanism. That one was about failure messages: a rule that referred to a name the search context could not resolve leaked that bare name into the substituted expression, where it became a SymbDC that failed with an anonymous "Cannot find Item X". Progress reports get their item name somewhere else entirely, so this issue was still open. It is now fixed as well, and it turns out the two halves of the title had two different causes.

## Where the name in a progress report comes from

A calculating operator prefixes its progress lines with the name of the item the result belongs to, taken from the back reference that TreeItem::SetReferredItem installs on a cache result: the config item that refers to it. The reporting context of the thread cannot serve here, because these lines are written from tile worker threads, which do not carry the context of the thread that started them.

## The indirection half was already fixed

Putting that name in front of these reports landed in fc735d68 (April, on the 19.3.0 tree), for the buffer, polygon, connect and impedance_matrix operators. With a configuration that buffers the same data twice, once directly and once through an indirect expression, both variants name their item on the current build. The discriminator is not the indirection but whether the result is bound to a config item.

## What was still missing

An intermediate result has no such back reference and stayed nameless. In

```
attribute<uint32> nr_points (pc6) := sequence_element_count(bg_buffer_multi_polygon(pc6/geometry, 25.0, 16b));
```

the buffer is an operator argument, an anonymous cache item, so its progress lines named no item at all:

```
bg_buffer_multi_polygon: processed 11,709 / 65,536 sequences of tile 3 / 8
```

That is the "not always shown" of the title, and it is what a modeller hits whenever the expensive call sits inside a larger expression instead of on a line of its own.

## The fix

A DataController now also carries an origin item for messages. Argument calculations are started from within CallCalcResultImpl of the calculation that needs them, all on the meta thread, so the origin travels down that nesting through one thread-local and is stored on each participating DataController before its operator is scheduled, to run on a worker thread later. A nameless intermediate then reports the config item whose calculation it is part of. The reference is weak, like the back reference itself: a name for a message must not keep an item alive. When several config items share one intermediate result, the name is that of the calculation that asked for it first.

The ten operator sites that formatted the prefix themselves now go through one GetProgressPrefix(), and the failure context in OperationContext ("while calculating X") uses the same name.

One thing on top: reportD appends the reporting context to any message of severity MajorTrace and up, so a progress line emitted on the thread that holds the context handle carried its item name twice. It is now appended only when the message does not already start with it.

## Verification

On the CBS postcode-6 polygons of 2022, read through GDAL, buffered once as a bound item and once inline:

```
[[/buf_direct]] bg_buffer_multi_polygon: processed 21,732 / 30,000 sequences of tile 0 / 1
[[/nr_points]]  bg_buffer_multi_polygon: processed 24,075 / 30,000 sequences of tile 0 / 1
```

Both name their item, once. Same result for a synthetic configuration with three variants (direct, indirect, inline) over 500,000 polygons in 8 tiles. The testcase battery runs 209 cases, 0 bad.

The configuration in the issue itself could not be rerun: the source data it reads under E:\SourceData\RSopen is no longer on the machine, and the configuration uses constructs that the current GeoDMS rejects (the obsolete subset operator, and an arrow reference to org_rel that no longer resolves). Reproducing it would mean migrating that configuration rather than testing the fix, hence the equivalent configurations above.

Fixed in d9379516 on the lookahead-scheduling branch.
